src/components/application_manager/src/commands/mobile/set_audio_streaming_indicator_request.cc

```diff
@@ -0,0 +1,124 @@
+ POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "application_manager/commands/mobile/set_audio_streaming_indicator_request.h"
+
+#include "application_manager/application.h"
+#include "application_manager/message_helper.h"
+
+namespace application_manager {
+
+namespace commands {
+
+SetAudioStreamingIndicatorRequest::SetAudioStreamingIndicatorRequest(
+    const MessageSharedPtr& message, ApplicationManager& application_manager)
+    : CommandRequestImpl(message, application_manager) {}
+
+SetAudioStreamingIndicatorRequest::~SetAudioStreamingIndicatorRequest() {}
+
+void SetAudioStreamingIndicatorRequest::Run() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  ApplicationSharedPtr app = application_manager_.application(connection_key());
+  if (!app) {
+    SendResponse(false, mobile_apis::Result::APPLICATION_NOT_REGISTERED);
+    LOG4CXX_ERROR(logger_, "Application is not registered");
+    return;
+  }
+
+  if (!app->is_media_application()) {
+    LOG4CXX_INFO(logger_, "Application is not media");
+    SendResponse(false, mobile_apis::Result::REJECTED);
+    return;
+  }
+
+  if (app->audio_streaming_indicator() ==
+      static_cast<mobile_apis::AudioStreamingIndicator::eType>(
+          (*message_)[strings::msg_params][strings::audio_streaming_indicator]
+              .asInt())) {
+    LOG4CXX_INFO(logger_, "Application is subscribed to this indicator");
+    SendResponse(false, mobile_apis::Result::IGNORED);
+    return;
+  }
+
+  smart_objects::SmartObject msg_params =
+      smart_objects::SmartObject(smart_objects::SmartType_Map);
+  msg_params = (*message_)[strings::msg_params];
+
+  SendHMIRequest(
+      hmi_apis::FunctionID::UI_SetAudioStreamingIndicator, &msg_params, true);
+}
+
+void SetAudioStreamingIndicatorRequest::on_event(
+    const event_engine::Event& event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  const smart_objects::SmartObject& message = event.smart_object();
+
+  switch (event.id()) {
+    case hmi_apis::FunctionID::UI_SetAudioStreamingIndicator: {
+      LOG4CXX_INFO(logger_, "Received UI_SetAudioStreamingIndicator event");
+
+      const hmi_apis::Common_Result::eType result_code =
+          static_cast<hmi_apis::Common_Result::eType>(
+              message[strings::params][hmi_response::code].asInt());
+      std::string response_info;
+      GetInfo(message, response_info);
+      const bool result = PrepareResultForMobileResponse(
+          result_code, HmiInterfaces::HMI_INTERFACE_UI);
+
+      if (result) {
+        ApplicationSharedPtr app =
+            application_manager_.application(connection_key());
+        if (app) {
+          app->set_audio_streaming_indicator(
+              static_cast<mobile_apis::AudioStreamingIndicator::eType>(
+                  (*message_)[strings::msg_params]
+                             [strings::audio_streaming_indicator].asInt()));
+        }
+      }
+      SendResponse(result,
+                   MessageHelper::HMIToMobileResult(result_code),
```

Comment:
@LuxoftAKutsan I agree with you but this moving out from scope of my CRQ. It looks like micro refactoring.